api/src/domain/tax-filings/taxFilingsFactor.test.ts

```diff
@@ -0,0 +1,240 @@
+      });
+
+      it("when lookup response is pending", async () => {
+        apiTaxFilingClient.lookup.mockResolvedValue({
+          state: "PENDING",
+          filings: [],
+        } as TaxFilingLookupResponse);
+        stubUserDataClient.get.mockResolvedValue(userData);
+        expect(await taxFilingClient.onboarding(userIdTaxIdBusinessName)).toEqual({
+          ...userData,
+          taxFilingData: { ...userData.taxFilingData, state: "PENDING" },
+        });
+        expect(apiTaxFilingClient.onboarding).not.toHaveBeenCalled();
+      });
+
+      it("when lookup response is success", async () => {
+        const filingData = generateTaxFiling({});
+        apiTaxFilingClient.lookup.mockResolvedValue({
+          state: "SUCCESS",
+          filings: [filingData],
+        } as TaxFilingLookupResponse);
+        stubUserDataClient.get.mockResolvedValue(userData);
+        expect(await taxFilingClient.onboarding(userIdTaxIdBusinessName)).toEqual({
+          ...userData,
+          taxFilingData: { ...userData.taxFilingData, state: "SUCCESS", filings: [filingData] },
+        });
+        expect(apiTaxFilingClient.onboarding).not.toHaveBeenCalled();
+      });
+    });
+
+    it("checks the onboarding api when lookup failed", async () => {
+      const userData = generateUserData({
+        taxFilingData: generateTaxFilingData({
+          state: undefined,
+          filings: [generateTaxFiling({})],
+        }),
+      });
+      apiTaxFilingClient.lookup.mockResolvedValue({
+        state: "FAILED",
+        filings: [],
+      } as TaxFilingLookupResponse);
+      apiTaxFilingClient.onboarding.mockResolvedValue({ state: "SUCCESS" } as TaxFilingOnboardingResponse);
+      stubUserDataClient.get.mockResolvedValue(userData);
+      expect(await taxFilingClient.onboarding(userIdTaxIdBusinessName)).toEqual({
+        ...userData,
+        taxFilingData: { ...userData.taxFilingData, state: "PENDING" },
+      });
+    });
+
+    describe("returns onboarding data when lookup failed", () => {
+      it("returns pending when onboarding is successful", async () => {
+        const userData = generateUserData({
+          taxFilingData: generateTaxFilingData({
+            state: undefined,
+            filings: [generateTaxFiling({})],
+          }),
+        });
+        apiTaxFilingClient.lookup.mockResolvedValue({
+          state: "FAILED",
+          filings: [],
+        } as TaxFilingLookupResponse);
+        apiTaxFilingClient.onboarding.mockResolvedValue({ state: "SUCCESS" } as TaxFilingOnboardingResponse);
+        stubUserDataClient.get.mockResolvedValue(userData);
+        expect(await taxFilingClient.onboarding(userIdTaxIdBusinessName)).toEqual({
+          ...userData,
+          taxFilingData: { ...userData.taxFilingData, state: "PENDING" },
+        });
+      });
+
+      describe("returns onboarding state when", () => {
+        it("responds with api error", async () => {
```

Comment:
similar comment as above, this is very confusing naming scheming because "it" (the object under test) is NOT the thing that responds with api error. one of its dependencies responds with an api error and _it_ handles / does something with that